File: redwood-yoga-java/src/main/java/io/github/orioncraftmc/meditate/internal/interfaces/YGMeasureFunc.java
package io.github.orioncraftmc.meditate.internal.interfaces;

import io.github.orioncraftmc.meditate.internal.YGNode;
import io.github.orioncraftmc.meditate.internal.YGSize;
import io.github.orioncraftmc.meditate.internal.enums.YGMeasureMode;
import org.jetbrains.annotations.NotNull;

@FunctionalInterface
public
interface YGMeasureFunc {
    @NotNull YGSize invoke(YGNode node, float width, YGMeasureMode widthMode, float height, YGMeasureMode heightMode);
}

File: redwood-yoga/src/commonMain/kotlin/app/cash/redwood/yoga/internal/interfaces/YGMeasureFunc.kt
package app.cash.redwood.yoga.internal.interfaces

import app.cash.redwood.yoga.internal.YGNode
import app.cash.redwood.yoga.internal.YGSize
import app.cash.redwood.yoga.internal.enums.YGMeasureMode

fun interface YGMeasureFunc {
    operator fun invoke(
      node: YGNode,
      width: Float,
      widthMode: YGMeasureMode,
      height: Float,
      heightMode: YGMeasureMode
    ): YGSize
}
